src/state.ml

```diff
@@ -264,24 +266,22 @@ let mtu config compress =
     | None -> 1500 (* TODO "client_merge_server_config" should do this! *)
     | Some x -> x
   in
-  let bs = match Config.find Cipher config with
-    | Some "AES-256-CBC" -> 16
-    | _ -> assert false
-  in
   (* padding, done on packet_id + compress + data *)
   let not_yet_padded_payload =
     4 (* packet id *) + if compress then 1 else 0
   in
   let hdrs =
-    3 (* hdr: 2 byte length, 1 byte op + key *) + bs (* IV *) + Packet.hmac_len
+    3 (* hdr: 2 byte length, 1 byte op + key *) +
+    Packet.block_size (* IV *) +
+    Packet.hmac_len
   in
   (* now we know: tun_mtu - hdrs is space we have for data *)
   let data = tun_mtu - hdrs in
```

Comment:
I guess this means we should ensure `tun_mtu > hdrs` in the config parser to avoid ending up with negative value here?

Reply:
yes, or rephrased: we should figure out a maximum / reasonable amount for `hdrs`, and check in the configuration parser that `Tun_mtu` (either in the configuration file, or transmitted via PUSH_REPLY) is way above that value. Since we're (atm?) concerned about tun interfaces only, we should in addition ensure that the remaining number is well above 20 (the IPv4 header size).

Reply:
I opened #51 to track this issue. Thanks for commenting on this function, in 56de7cf I added conditionals for static keys (where the header includes a timestamp, but excludes the 1 byte operation and key id).